### ISSUE TITLE
Using an enum for HTTP method is problematic

### ISSUE THREAD

**Issue by mheath** [cloudfoundry/dropsonde-protocol#1]
I had an email conversation with @ajackson about this and wanted to create a GitHub issue to open up the discussion further.

Currently, the HTTP events use an enum to hold the HTTP method, https://github.com/cloudfoundry/dropsonde-protocol/blob/master/events/http.proto#L12-L18.

This is problematic because HTTP methods are not limited to a fixed set. An HTTP method can be an arbitrary case sensitive ASCII string. (See 'extension-method' in the HTTP/1.1 RFC http://www.w3.org/Protocols/rfc2616/rfc2616-sec5.html#sec5.1.1.)

Examples of standard extension-methods include the WebDAV protocol which adds 7 HTTP methods and RFC 5789 adds the PATCH method.

The challenge now is that the method fields on the HTTP events is marked as required so it can't be removed and maintain backwards compatibility.

So, how do we create HTTP Dropsonde events for non-standard HTTP methods?

**Comment by dsabeti** [comment]
Hi @mheath 

This is a really good point, and we've verified that a PATCH request results in an HTTPStartStopEvent with Method=0 (no value) because PATCH is not currently in the Enum.

The problem with changing Method from an Enum to a string is that we will have to create a migration plan, including deprecating the existing Method field, sending a duplicate field with type String, etc. The team is already discussing some changes to the protocol that will require a migration plan on its own. So we're going to punt on changing the type of the Method field until we get around to this larger migration.

In the meantime, we can always add widely-accepted HTTP Methods to the existing Enum. This change would not be backwards incompatible and would require no special migration. [We've added a story](https://www.pivotaltracker.com/story/show/102264456) to our tracker to add in PATCH and the WebDAV methods.
